internal/controller/progressive/progressive.go

```diff
@@ -99,10 +104,12 @@ func ProcessResource(
 			return false, false, 0, fmt.Errorf("error getting %s: %v", currentUpgradingChildDef.GetKind(), err)
 		}
 		if currentUpgradingChildDef == nil {
-			if liveRolloutObject.GetPromotedChildStatus() == nil {
-				rolloutObject.SetPromotedChildStatus(&apiv1.PromotedChildStatus{Name: existingPromotedChild.GetName()})
-			} else {
-				rolloutObject.SetPromotedChildStatus(liveRolloutObject.GetPromotedChildStatus().DeepCopy())
+			promotedChildStatus := rolloutObject.GetPromotedChildStatus()
+			if promotedChildStatus == nil || promotedChildStatus.Name != existingPromotedChild.GetName() {
+				err = rolloutObject.ResetPromotedChildStatus(existingPromotedChild)
+				if err != nil {
+					return false, false, 0, err
+				}
```

Comment:
I figure we need to also Reset this in the case that the promoted child name has changed from before

Reply:
Also, by calling `ResetPromotedChildStatus()`, we now reset the entire encapsulating promoted child struct for that particular Rollout.